src/Command/Source/SourceDownloadCoversCommand.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @file
+ * Helper command to search the source table for potential new images.
+ */
+
+namespace App\Command\Source;
+
+use App\Entity\Source;
+use App\Event\VendorEvent;
+use App\Service\VendorService\ProgressBarTrait;
+use App\Service\VendorService\VendorImageValidatorService;
+use App\Utils\CoverVendor\VendorImageItem;
+use App\Utils\Types\VendorState;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class SourceDownloadCoversCommand extends Command
+{
+    use ProgressBarTrait;
+
+    protected static $defaultName = 'app:source:download';
+
+    private $em;
+    private $validator;
+    private $dispatcher;
+
+    public function __construct(EntityManagerInterface $entityManager, VendorImageValidatorService $validator, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->em = $entityManager;
+        $this->validator = $validator;
+        $this->dispatcher = $eventDispatcher;
+
+        parent::__construct();
+    }
+
+    /**
+     * Define the command.
+     */
+    protected function configure()
+    {
+        $this->setDescription('Try to (re-)download source records that have not been downloaded into Cover store')
+            ->addOption('vendor-id', null, InputOption::VALUE_OPTIONAL, 'Limit to this vendor')
+            ->addOption('identifier', null, InputOption::VALUE_OPTIONAL, 'Only for this identifier');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $vendorId = $input->getOption('vendor-id');
+        $identifier = $input->getOption('identifier');
+
+        $batchSize = 50;
```

Comment:
Would prefer a class constant, but given that it is just a command it's not important

Reply:
`Constants defined using the const keyword must be declared at the top-level scope`

Reply:
Don't what it in top level.

Reply:
👍